src/types.rs

```diff
@@ -360,3 +389,49 @@ impl Internal for InvalidRequest {
         }
     }
 }
+
+// ---
+// JSON Serialization for Tiles
+// Since we can't directly add these derives on the proto_types
+// ---
+// Just trust that these values are OK
+#[derive(Serialize, Deserialize)]
+pub struct SerializableSquare {
+    pub north_wall: i32,
+    pub east_wall: i32,
+    pub south_wall: i32,
+    pub west_wall: i32,
+    pub square_type: i32,
+}
+
+impl From<Square> for SerializableSquare {
+    fn from(item: Square) -> Self {
+        SerializableSquare {
+            north_wall: i32::from(item.north_wall),
+            east_wall: i32::from(item.east_wall),
+            south_wall: i32::from(item.south_wall),
+            west_wall: i32::from(item.west_wall),
+            square_type: i32::from(item.square_type),
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct SerializableTile {
+    pub position: MapPosition,
+    pub squares: Vec<SerializableSquare>,
+}
+
+impl From<Tile> for SerializableTile {
+    fn from(item: Tile) -> Self {
+        let mut serializable_squares = vec![];
+        for square in item.squares {
+            serializable_squares.push(SerializableSquare::from(square));
+        }
+        SerializableTile {
+            squares: serializable_squares,
+            position: item.position,
+        }
+    }
+}
+// ---
```

Comment:
killlllll

Reply:
it's my _very_ thoughtful comment section ended. but i don't care particularly strongly about keeping the sectioning or not /shrug.keeping for now complain again if you really want it